trino/client.py

```diff
@@ -494,20 +498,63 @@ def __iter__(self):
             for row in rows:
                 self._rownumber += 1
                 logger.debug("row %s", row)
-                yield row
+                if not self._experimental_python_types:
+                    yield row
+                else:
+                    yield self._map_to_python_types(row, self._query.columns)
 
     @property
     def response_headers(self):
         return self._query.response_headers
 
+    @classmethod
+    def _map_to_python_type(cls, item: Tuple[Any, Dict]) -> Any:
+        (value, data_type) = item
+
+        if value is None:
+            return None
+
+        raw_type = data_type["typeSignature"]["rawType"]
+
+        try:
+            if isinstance(value, list):
+                raw_type = {
+                    "typeSignature": data_type["typeSignature"]["arguments"][0]["value"]
+                }
+                return [cls._map_to_python_type((array_item, raw_type)) for array_item in value]
+            elif "decimal" in raw_type:
+                return Decimal(value)
+            elif raw_type == "date":
+                return datetime.strptime(value, "%Y-%m-%d").date()
+            elif raw_type == "timestamp with time zone":
+                dt, tz = value.rsplit(' ', 1)
+                if tz.startswith('+') or tz.startswith('-'):
+                    return datetime.strptime(value, "%Y-%m-%d %H:%M:%S.%f %z")
+                return datetime.strptime(dt, "%Y-%m-%d %H:%M:%S.%f").replace(tzinfo=pytz.timezone(tz))
+            elif "timestamp" in raw_type:
+                return datetime.strptime(value, "%Y-%m-%d %H:%M:%S.%f")
+            elif "time with time zone" in raw_type:
+                return datetime.strptime(value, "%H:%M:%S.%f").time()
```

Comment:
Shouldn't this include timezone information? 

Note the `dateutil.parser.parse` method could be useful and replace the logic in lines 527–539, somewhat ignoring the specific `raw_type`, i.e., `value` should contain the timezone information if the `raw_type` is `timestamp with time zone` or `time with time zone`, i.e., 


```python
from dateutil.parser import parse

if raw_type in ("time",  "time with time zone", "timestamp", "timestamp with time zone"):
    result = parse(value)
    
    if raw_type in ("time",  "time with time zone"):
        result = result.time().replace(tzinfo=result.tzinfo)
        
    return result
```